core/src/main/java/jenkins/security/s2m/DefaultFilePathFilter.java

```diff
@@ -59,6 +62,15 @@ protected boolean op(String op, File f) throws SecurityException {
                     LOGGER.log(Level.FINE, "agent allowed to {0} {1}", new Object[] {op, f});
                     return true;
                 } else {
+                    try {
+                        Field current = Request.class.getDeclaredField("CURRENT");
+                        current.setAccessible(true);
+                        Field createdAt = Command.class.getDeclaredField("createdAt");
+                        createdAt.setAccessible(true);
+                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller " + op + " on " + f, (Throwable) createdAt.get(((ThreadLocal) current.get(null)).get()));
```

Comment:
```suggestion
                        LOGGER.log(Level.WARNING, "Permitting agent-to-controller '" + op + "' on '" + f  + "' which will be removed soon. See https://www.jenkins.io/redirect/permitted-agent-to-controller-file-access", (Throwable) createdAt.get(((ThreadLocal) current.get(null)).get()));
```

Makes this actionable to admins. We'll need to provide some instructions how to identify the culprit, best done on the site, and ask that a bug be reported.

Reply:
https://www.jenkins.io/redirect/permitted-agent-to-controller-file-access is a 404.